packages/edit-site/src/components/global-styles/color-panel.js

```diff
@@ -20,59 +21,67 @@ export function useHasColorPanel( { supports } ) {
 	);
 }
 
-export default function ColorPanel( {
-	context: { supports, name },
-	getStyle,
-	setStyle,
-} ) {
-	const solids = useSetting( 'color.palette', name );
-	const gradients = useSetting( 'color.gradients', name );
-	const areCustomSolidsEnabled = useSetting( 'color.custom', name );
-	const areCustomGradientsEnabled = useSetting(
+export default function ColorPanel( { name } ) {
+	const supports = getSupportedGlobalStylesPanels( name );
+	const [ solids ] = useSetting( 'color.palette', name );
+	const [ gradients ] = useSetting( 'color.gradients', name );
+	const [ areCustomSolidsEnabled ] = useSetting( 'color.custom', name );
+	const [ areCustomGradientsEnabled ] = useSetting(
 		'color.customGradient',
 		name
 	);
-	const isLinkEnabled = useSetting( 'color.link', name );
-	const isTextEnabled = useSetting( 'color.text', name );
-	const isBackgroundEnabled = useSetting( 'color.background', name );
 
 	const hasLinkColor =
 		supports.includes( 'linkColor' ) &&
-		isLinkEnabled &&
 		( solids.length > 0 || areCustomSolidsEnabled );
 	const hasTextColor =
 		supports.includes( 'color' ) &&
-		isTextEnabled &&
 		( solids.length > 0 || areCustomSolidsEnabled );
 	const hasBackgroundColor =
 		supports.includes( 'backgroundColor' ) &&
-		isBackgroundEnabled &&
```

Comment:
I removed these checks because they were making the color panel unavailable when it should be available? I guess we have a bug somewhere?

Reply:
I also noticed that the "custom color" links are not visible. I guess the `useSetting` calls are not working properly for some reason, I'll try to find out why.

Reply:
Solved in c8296e3